src/test/java/com/example/cats/service/ProductServiceImplTest.java

```diff
@@ -0,0 +1,90 @@
+package com.example.cats.service;
+
+import com.example.cats.domain.Category;
+import com.example.cats.domain.Product;
+import com.example.cats.dto.ProductDTO;
+import com.example.cats.mapper.ProductMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mapstruct.factory.Mappers;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ProductServiceImplTest {
+
+    private ProductServiceImpl service;
+
+    @BeforeEach
+    void setUp() {
+        ProductMapper mapper = Mappers.getMapper(ProductMapper.class);
```

Comment:
you shoudn't get your mapper before each test, please re-watch the code from lecture, you can use the @SpringBootTest with limited scope and you can import mapper configuration